EndlessClient/Dialogs/GuildDialog.cs

```diff
@@ -1,233 +1,522 @@
+                    var recruiterName = dlgJoin.Responses[1];
+
+                    if (guildTag.Length == 0)
+                    {
+                        e.Cancel = true;
+                        var dlg = _messageBoxFactory.CreateMessageBox(DialogResourceID.GUILD_CREATE_TAG_FIELD_EMPTY);
+                        dlg.Show();
+                        return;
+                    }
+
+                    if (recruiterName.Length == 0)
+                    {
+                        e.Cancel = true;
+                        var dlg = _messageBoxFactory.CreateMessageBox(DialogResourceID.GUILD_RECRUITER_INPUT_MISSING);
+                        dlg.Show();
+                        return;
+                    }
+
+                    if (guildTag.Length == 1)
+                    {
+                        e.Cancel = true;
+                        var dlg = _messageBoxFactory.CreateMessageBox(DialogResourceID.GUILD_CREATE_TAG_TOO_SHORT);
+                        dlg.Show();
+                        return;
+                    }
+
+                    _guildActions.RequestToJoinGuild(guildTag, recruiterName);
+                }
+            };
+
+            dlgJoin.Show();
+        }
+
+        private void ShowLeaveGuildMessageBox()
+        {
+            var dlgLeave = _messageBoxFactory.CreateMessageBox(DialogResourceID.GUILD_PROMPT_LEAVE_GUILD, whichButtons: EODialogButtons.OkCancel);
+            dlgLeave.DialogClosing += (_, e) =>
+            {
+                if (e.Result == XNADialogResult.OK)
+                {
+                    _guildActions.LeaveGuild();
+                }
+            };
+
+            dlgLeave.Show();
+
+        }
+
+        private void ShowRegisterGuildMessageBox()
+        {
+            var dlgRegister = _textMultiInputDialogFactory.Create(
+                _localizedStringFinder.GetString(EOResourceID.GUILD_REGISTER_GUILD),
+                _localizedStringFinder.GetString(EOResourceID.GUILD_ENTER_YOUR_GUILD_DETAILS),
+                TextMultiInputDialog.DialogSize.Three,
+                new TextMultiInputDialog.InputInfo { Label = _localizedStringFinder.GetString(EOResourceID.GUILD_GUILD_TAG), MaxChars = 3 },
+                new TextMultiInputDialog.InputInfo { Label = _localizedStringFinder.GetString(EOResourceID.GUILD_GUILD_NAME), MaxChars = 24 },
+                new TextMultiInputDialog.InputInfo { Label = _localizedStringFinder.GetString(EOResourceID.GUILD_GUILD_DESCRIPTION), MaxChars = 240 }
+            );
+
+            dlgRegister.DialogClosing += (_, e) =>
+            {
+                if (e.Result == XNADialogResult.OK)
+                {
+                    var guildTag = dlgRegister.Responses[0];
+                    var guildName = dlgRegister.Responses[1];
+                    var guildDescription = dlgRegister.Responses[2];
+
+                    if (!_characterInventoryProvider.ItemInventory.Any(x => x.ItemID == 1 && x.Amount >= 50_000))
```

Comment:
Is this hard-coded into the vanilla client too? I thought there was an eoserv config option to change the guild create cost requirement, but maybe that was one of the "use a custom client with this setting" options.

Reply:
The client validates 50k and there's no way for the server to configure a price

Reply:
Blehhh. Classic. I'm not surprised but for some reason I'm disappointed.